lucille-core/src/main/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnector.java

```diff
@@ -255,16 +254,14 @@ private void iterateOtherSQL(ResultSet rs2, String[] columns2, Document doc, Int
     }
 
     do {
-      // Convert to do-while I think we can avoid the rs2.previous() call.
-      // TODO: support non INT primary key
-      int otherJoinId = rs2.getInt(joinField);
-      if (otherJoinId < joinId) {
+      Object otherJoinId = rs2.getObject(joinField);
+      // will throw error if different class, if either of objects are null or not Comparable
+      int comparison = ((Comparable) otherJoinId).compareTo(joinId);

```

Comment:
Would probably be helpful to catch each of these specific exceptions and then throw a ConnectorException with the original exception and a message explaining what went wrong.

Reply:
implemented catching for all exceptions in [91eec79](https://github.com/kmwtechnology/lucille/pull/184/commits/91eec79718ff3de576700b2d2911d5c1515ea985)